stl/inc/ratio

```diff
@@ -19,20 +19,22 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
-template <intmax_t _Val>
-struct _Abs : integral_constant<intmax_t, (_Val < 0 ? -_Val : _Val)> {}; // computes absolute value of _Val
+_NODISCARD constexpr intmax_t _Abs(const intmax_t _Val) noexcept {
+    return _Val < 0 ? -_Val : _Val;
+}
 
 template <intmax_t _Ax, intmax_t _Bx, bool _Sfinae = false,
-    bool _Good = (_Abs<_Ax>::value <= INTMAX_MAX / (_Bx == 0 ? 1 : _Abs<_Bx>::value))>
+    bool _Good = (_Abs(_Ax) <= INTMAX_MAX / (_Bx == 0 ? 1 : _Abs(_Bx)))>
 struct _Safe_mult : integral_constant<intmax_t, _Ax * _Bx> {}; // computes _Ax * _Bx without overflow
 
 template <intmax_t _Ax, intmax_t _Bx, bool _Sfinae>
 struct _Safe_mult<_Ax, _Bx, _Sfinae, false> { // _Ax * _Bx would overflow
     static_assert(_Sfinae, "integer arithmetic overflow");
 };
 
-template <intmax_t _Val>
-struct _Sign_of : integral_constant<intmax_t, (_Val < 0 ? -1 : 1)> {}; // computes sign of _Val
+_NODISCARD constexpr intmax_t _Sign_of(const intmax_t _Val) noexcept {
+    return _Val < 0 ? -1 : 1;
+}
 
 template <intmax_t _Ax, intmax_t _Bx, bool _Good, bool _Also_good>
 struct _Safe_addX : integral_constant<intmax_t, _Ax + _Bx> {}; // computes _Ax + _Bx without overflow

```

Comment:
`_Safe_addX` is static asserting that it is valid, so I think we should be able to port that also over to a function

Reply:
How?

Reply:
Shouldnt it be something like
```cpp
_NODISCARD constexpr intmax_t _Safe_add(const intmax_t _Ax, const _intmax_t _Bx) noexcept {
    if (_Sign_of(_Ax) != _Sign_of(_Bx) && _Abs(_Ax) <= INTMAX_MAX - _Abs(_Bx)) {
        static_assert(_Always_false<intmax_t >, "integer arithmetic overflow");
    }

    return _Ax + _Bx;
}
```

Reply:
That's not possible, because that'd have to be written with `if constexpr`, which requires the parameters to be NTTPs, kind of destroying the purpose of this PR. Technically, it'll be written in function form, but still...

Reply:
Implemented 👍